### PR TITLE
feat: caip10 ethereum+eos integration test

### PR DESCRIPTION
**Pull request by ukstv**
Note: Consider it as a part of the whole caip10-integration thing.

Add caip10-link integration test, starting with ethereum.

Suggestions are welcome on renaming canary-integration.

One idea I have in mind is to run `canary-integration` only on master by changing `test` script to `test:full` inside the package, and adding a CircleCI rule that runs `test:full` on master only.

### CODE REVIEW

File: packages/canary-integration/src/create-ipsf.ts [hunk 1]
```diff
@@ -0,0 +1,77 @@
+import dagJose from 'dag-jose'
```
Comment (stbrody) [inline packages/canary-integration/src/create-ipsf.ts:1]: typo in filename: create-ipsf

**Review by stbrody — COMMENTED** [review]
Why would we only run this on master branch?  That would mean its only tested after a release is already made.  Why not run these tests on every commit like any other test?

File: packages/canary-integration/src/create-ipsf.ts (hunk 1 above)
Comment (ukstv) [reply packages/canary-integration/src/create-ipsf.ts]: Oops. Thanks!

**Comment by ukstv** [comment]
> That would mean its only tested after a release is already made.

Yes, this is not good. I was thinking about conserving runners capacity, be it on CircleCI or GitHub Actions. Maybe not on master. Maybe better go with a button-initiated workflow, like for releases.

**Comment by stbrody** [comment]
>  I was thinking about conserving runners capacity, be it on CircleCI or GitHub Actions. 

Yeah, personally I'd vote for running this all the time like all our other tests, at least for now.  If/when in the future our test execution times are too long, or our costs associated with running the tests is too high, then we can take a holistic look on how to break up tests to address it.  By that time, we may have many more tests which may be even slower or more expensive to run than these are, so this kind of feels like premature optimization to me.

File: packages/canary-integration/src/create-ceramic.ts [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  const threeIdResolver = ThreeIdResolver.getResolver(ceramic)
+  const resolver = new Resolver({
+    ...threeIdResolver,
+    ...keyDidResolver,
+  })
+  const did = new DID({ provider, resolver })
+  await ceramic.setDID(did)
+  await did.authenticate()
+
+  return ceramic
```
Comment (oed) [inline packages/canary-integration/src/create-ceramic.ts:37]: FWIW, @PaulLeCam created a `jest-environment-ceramic` package. At some point we should migrate that package to the js-ceramic code base and deduplicate a bunch of code.

**Review by oed — APPROVED** [review]
lgtm!

Sorry for taking so long to review this.

**Pull request merged**